Update .NET 10 documentation with comprehensive feature integration and address review feedback (#47938)
dotnet/docs: docs/core/whats-new/dotnet-10/sdk.md

For more information about managing .NET tools, see [Manage .NET tools](../../tools/global-tools.md).

### Use the `any` RuntimeIdentifier with platform-specific .NET tools

The [platform-specific .NET tools](#platform-specific-net-tools) feature is great for making sure tools are optimized for specific platforms that you target ahead-of-time. However, there are times where you won't know all of the platforms that you'd like to target, or sometimes .NET itself will learn how to support a new platform, and you'd like your tool to be runnable there too.

> .NET is great at this - the platform at its heart is meant to support this kind of platform-agnostic execution. To make platform-specific .NET tools work this way, you only need to add one thing to your project file: the `any` Runtime Identifier.

Reviewer: ```suggestion
To make your tool work this way, add the `any` runtime identifier to your project file:
```

```xml
<PropertyGroup>
  <RuntimeIdentifiers>
       linux-x64;
       linux-arm64;
       macos-arm64;
       win-x64;
       win-arm64;
       any
  </RuntimeIdentifiers>
</PropertyGroup>
```

> This RuntimeIdentifier is at the 'root' of the platform-compatibility checking, and since it declares support for _any_ platform, the tool that gets packaged will be the most compatible kind of tool - a framework-dependent, platform-agnostic .NET dll, which requires a compatible .NET Runtime to execute. When you perform a `dotnet pack` to create your tool, you'll see a new package for the `any` RuntimeIdentifier appear alongside the other platform-specific packages and the top-level manifest package.

Reviewer: ```suggestion
This RuntimeIdentifier is at the 'root' of the platform-compatibility checking, and since it declares support for _any_ platform, the tool that gets packaged will be the most compatible kind of tool - a framework-dependent, platform-agnostic .NET DLL, which requires a compatible .NET Runtime to execute. When you perform a `dotnet pack` to create your tool, you'll see a new package for the `any` RuntimeIdentifier appear alongside the other platform-specific packages and the top-level manifest package.
```

> This is the exact same kind of tool that you would make in .NET 9 and earlier, but now it fits into the overall goal of enabling platform-specific .NET tools!
> 

Reviewer: ```suggestion
```

The [platform-specific .NET tools](#platform-specific-net-tools) feature is great for making sure tools are optimized for specific platforms that you target ahead-of-time. However, there are times where you won't know all of the platforms that you'd like to target, or sometimes .NET itself will learn how to support a new platform, and you'd like your tool to be runnable there too.

To make your tool work this way, add the `any` runtime identifier to your project file:

```xml
<PropertyGroup>
  <RuntimeIdentifiers>
       linux-x64;
       linux-arm64;
       macos-arm64;
       win-x64;
       win-arm64;
       any
  </RuntimeIdentifiers>
</PropertyGroup>
```

This RuntimeIdentifier is at the 'root' of the platform-compatibility checking, and since it declares support for _any_ platform, the tool that gets packaged will be the most compatible kind of tool - a framework-dependent, platform-agnostic .NET DLL, which requires a compatible .NET Runtime to execute. When you perform a `dotnet pack` to create your tool, you'll see a new package for the `any` RuntimeIdentifier appear alongside the other platform-specific packages and the top-level manifest package.

> ### CLI introspection with `--cli-schema`

Reviewer 2: ```suggestion

### CLI introspection with `--cli-schema`

```

```

For more details, see [Testing with `dotnet test`](../../testing/unit-testing-with-dotnet-test.md).

> ## Preview 7 additions

Reviewer: Don't mention preview versions at all.

The following feature was added in Preview 7:

> 

Reviewer 2: Remove this header so that the Preview 7 features are incorporated into the main article.

> ### Use the `any` RuntimeIdentifier with platform-specific .NET tools

Reviewer 2: Move this section under the H2 for ".NET tool enhancements"

The [platform-specific .NET tools](#platform-specific-net-tools) feature released in Preview 6 is great for making sure your tools are optimized for specific platforms that you target ahead-of-time. However, there are times where you won't know all of the platforms that you'd like to target, or sometimes .NET itself will learn how to support a new platform, and you'd like your tool to be runnable there too.

The good news is that .NET is great at this - the platform at its heart is meant to support this kind of platform-agnostic execution. To make your new platform-specific .NET tools work this way, you only need to add one thing to your project file: the `any` Runtime Identifier.

```diff
<PropertyGroup>
  <RuntimeIdentifiers>
        linux-x64;
        linux-arm64;
        macos-arm64;
        win-x64;
-       win-arm64
+       win-arm64;
+       any
  </RuntimeIdentifiers>
</PropertyGroup>
```

> This RuntimeIdentifier is at the 'root' of our platform-compatibility checking, and since it declares support for, well, _any_ platform, the tool that we package for you will be the most compatible kind of tool - a framework-dependent, platform-agnostic .NET dll, which requires a compatible .NET Runtime in order to execute. When you perform a `dotnet pack` to create your tool, you'll see a new package for the `any` RuntimeIdentifier appear alongside the other platform-specific packages and the top-level manifest package.

Reviewer: Don't use "our" or "we"

Reviewer: "in order to" -> "to"

> The eagle-eyed among you will note that this is the exact same kind of tool that you would make in .NET 9 and earlier, but now it fits into the overall goal of enabling platform-specific .NET tools!

Reviewer: This sounds more markety than docs-like.
